### ISSUE TITLE
Double conditional field values not saved correctly

### ISSUE THREAD

**Issue by jsmrtn** [verbb/formie#620]
**Description**

We have a contact form which has preferred contact field with a checkbox group. Clicking each one of the checkboxes  reveals a hidden field related to each checkbox.

It uses native templates and functionality, just some light CSS adjustments. The JS works and reveals the two panels, however, if you check both checkboxes and data is submitted to both the Email and Telephone field then this data is not recorded. If you check one box and only add content that exposed field then the data is recorded to the db.

**Steps to reproduce**
1. Create a form that has a checkbox to conditionally show/hide fields

<img width="1022" alt="Screenshot 2021-10-26 at 10 57 07" src="https://user-images.githubusercontent.com/1694957/138856469-a41e6c3d-e7b1-48e6-bf46-729b895f6a48.png">
<img width="1086" alt="Screenshot 2021-10-26 at 10 57 17" src="https://user-images.githubusercontent.com/1694957/138856434-6b098da5-a9d3-4454-835e-3b5dfce60a01.png">

<img width="1071" alt="Screenshot 2021-10-26 at 10 57 24" src="https://user-images.githubusercontent.com/1694957/138856443-296476ca-059a-49e4-847a-4538ea9dfd2a.png">

2. Try to submit form with both boxes checked and both values filled and note the submission does not contain either value

<img width="571" alt="Screenshot 2021-10-26 at 10 59 18" src="https://user-images.githubusercontent.com/1694957/138856537-92fb09dd-7f18-420b-a821-756057d0db55.png">

<img width="1037" alt="Screenshot 2021-10-26 at 11 03 18" src="https://user-images.githubusercontent.com/1694957/138856635-99837b3d-cee8-4b70-a234-67aa3990f1b5.png">

### Form settings
Please pick from the options below.

- Multi-page form: No
- Submission Method: Page Reload
- Client-side Validation: No
- Custom Form Templates: No

**Additional info**
- Plugin version: 1.4.17
- Craft version: 3.7.13
- Multi-site: No

**Comment by engram-design** [comment]
Yeah, we've come across this before with https://github.com/verbb/formie/issues/477 where comparing with checkboxes is a bit tricky, and not _quite_ sure the best way to handle it.

Technically the `is` operator is a straight `==` or `equal` comparer. The value for `Preferred contact method` is technically `["E-mail", "Phone"]`.

So when comparing, it's checking if `["E-mail", "Phone"] == "E-mail"` which is false, and therefore classified as hidden. According to the change in https://github.com/verbb/formie/issues/504 we exclude any field value - even if provided in the payload - for conditionally hidden fields.

The recommended approach is to change your condition to `contains`. However it's also technically a bug that the front-end acts in a different manner than the back-end. Both fields should actually be hidden if you have both checkboxes checked, because of the equality comparison described above.

I have scheduled a re-look at this in 1.5.x to provide a better UI for this field fields that deal with multiple values.

**Comment by engram-design** [comment]
After some more careful thought I've modified the behaviour server-side to fix this issue.

Firstly, I don't believe a UI change needs to be made, because a single line in the conditions table should equate to a single condition. If you wanted to check against multiple values, you can create multiple conditions. So as such, the front-end behaviour is correct.

If you wanted to check if **both** the "Email" and "Phone" checkboxes were checked, that's two rows in the conditions table. You could even change "All" or "Any" depending if you need to `and/or` conditions. For whatever reason, if you wanted a field to only show when both options (or either options) were checked, that's the way to go about it.

Fixed for the next release. To get this fix early, change your `verbb/formie` requirement in `composer.json` to:

```
"require": {
  "verbb/formie": "dev-craft-3 as 1.4.22",
  "...": "..."
}
```

Then run `composer update`.

**Comment by jsmrtn** [comment]
Thanks Josh. Completely understand the logic of it matching on both being ticked, but I suggest the way you've handled it is best. In my mind, users are going to use it in the way we've defined (that is, 'Checkbox equals X so show Y field'), so allowing multiple checkboxes to show (and save) multiple fields is the way forwards. 

Appreciate the quick turnaround ✌🏽

**Comment by engram-design** [comment]
Fixed in [1.4.23](https://github.com/verbb/formie/releases/tag/1.4.23)

**Comment by simonleadbetter** [comment]
Hi Josh. My colleague opened this ticket with you and the patch did fix the issue dev-craft-3 as 1.4.22, however, I have tried to update Formie to the latest release (1.4.23) and I see the following error message:

`Setting unknown property: verbb\formie\fields\formfields\Checkboxes::rowUid`

Downgrading to 1.4.21 then upgrading didn’t work.

Any ideas what I would need to do to remedy this?

**Comment by engram-design** [comment]
When does this error occur? During the update, when editing a form, etc? The `rowUid` property isn't new, and inherited by all fields, so it's certainly not an unknown property.

**Comment by simonleadbetter** [comment]
It occurs post update and creates an Internal Server Error so Craft control panel is inaccessible as well as the front-end.

Here’s the Formie log in case it is useful…

`2021-11-03 04:59:50 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\Address::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 04:59:50 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\MultiLineText::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 04:59:50 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\Email::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 04:59:50 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\SingleLineText::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 04:59:50 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\Hidden::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 04:59:50 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\Hidden::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 04:59:50 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\Hidden::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 04:59:50 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\SingleLineText::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 04:59:50 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\Address::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 04:59:50 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\MultiLineText::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 04:59:50 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\Email::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 04:59:50 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\SingleLineText::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 04:59:50 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\Hidden::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 04:59:50 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\Hidden::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 04:59:50 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\Hidden::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 04:59:50 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\SingleLineText::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 04:59:44 [-][-][-][info][application] $_GET = [
    'p' => 'admin'
]

$_POST = []
2021-11-03 05:00:02 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\Address::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 05:00:02 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\MultiLineText::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 05:00:02 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\Email::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 05:00:02 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\SingleLineText::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 05:00:02 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\Hidden::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 05:00:02 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\Hidden::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 05:00:02 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\Hidden::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 05:00:02 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\SingleLineText::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 05:00:02 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\Address::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 05:00:02 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\MultiLineText::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 05:00:02 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\Email::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 05:00:02 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\SingleLineText::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 05:00:02 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\Hidden::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 05:00:02 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\Hidden::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 05:00:02 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\Hidden::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 05:00:02 [-][-][-][info][formie] Setting unknown property: verbb\formie\fields\formfields\SingleLineText::rowUid /Users/simonleadbetter/Documents/Websites/anatolia-tile-website/vendor/yiisoft/yii2/base/Component.php:210
2021-11-03 05:00:01 [-][-][-][info][application] $_GET = [
    'p' => 'admin/actions/users/session-info'
    'dontExtendSession' => '1'
]`

**Comment by engram-design** [comment]
Hmm. Have you run:
```
composer update
./craft migrate/all
./craft project-config/rebuild
```

It almost sounds like some files are missing from the component files.

I would also try clearing all caches:
```
./craft clear-caches/all
```

Maybe even completely clearing your `vendor` folder and `composer install`-ing. You could also clear your `storage/runtime` folder.

**Comment by simonleadbetter** [comment]
I did try all your suggestions but still no joy. Console displays this error…

`Rebuilding the project config from the current state ... error: Setting unknown property: verbb\formie\fields\formfields\Checkboxes::rowUid`

I did also delete project-config directory.

It’s so strange. Using dev-craft-3 as 1.4.22 or 1.4.21 works fine.

**Comment by engram-design** [comment]
Are you able to update to 1.4.23 (the latest release)? What PHP version are you running?

**Comment by simonleadbetter** [comment]
Unfortunately not. Same error. Locked to dev-craft-3 as 1.4.22 or 1.4.21.

Running 7.4.12 locally.

**Comment by engram-design** [comment]
So there must be some fields in your `fields` database table that contains `rowUid`. Are you able to have a look at some Formie fields in this database table and shoot them through?

Or, if you'd prefer, send your `fields` table to `web at verbb.io` for us to take a look at.

I would also do a search in your `config/project` folder for `rowUid` in case that's being stored in your YAML files (it shouldn't be).

**Comment by engram-design** [comment]
The only way I can replicate this is by removing `public $rowUid;` from the `FormFieldTrait.php`.

Are you able to navigate to `vendor/formie/src/base/FormFieldTrait.php` and search for `rowUid` (should be line 70), just to confirm that it **is** there?

**Comment by engram-design** [comment]
Fixed in [1.4.24](https://github.com/verbb/formie/releases/tag/1.4.24)
